build_dependency_tree.py

```diff
@@ -3,14 +3,27 @@
 import sys
 import re
 from pathlib import Path
+from typing import Optional
 from typing import List
+from typing import Union
 
-args = [arg for arg in sys.argv]
-files_in_dir = [
-    [file for file in Path(root).glob('*') # TODO: Make recursive search 
-     if file.suffix in ('h', 'hpp', 'c', 'cpp')]
-    for root in args
-]
+args: List[Union[bytes, str]] = [arg for arg in sys.argv]
+files_in_project = []
+
+
+def extract_files_from_directory(
+        dir_or_file: Path, list_o_files: List[Optional[Path]]
+):
```

Comment:
```python
def extract_files_from_directory(
    dir_or_file: Path,
    list_of_files: List[Optional[Path]]=[]
):
```
Is what I suggested her with the comment you resolved above

Reply:
Why is it `List[Optional[Path]]`? Specifically the Optional

Reply:
I simply added the empty initialization. I personally think it should be a union of Str and Path

Reply:
Ahh, I just realized my latest push isn't actually up